src/mam4xx/nucleation.hpp

```diff
@@ -317,7 +319,554 @@ void mer07_veh02_wang08_nuc_1box(int newnuc_method_user_choice,
     // ratenuclt is #/cm3/s; dnclusterdt is #/m3/s
     dnclusterdt = exp(rateloge) * 1.0e6;
   }
-}
+} // end mer07_veh02_wang08_nuc_1box()
+
+//-----------------------------------------------------------------------------
+// Calculates new particle production from homogeneous nucleation
+// using nucleation rates from either
+// Merikanto et al. (2007) h2so4-nh3-h2o ternary parameterization
+// Vehkamaki et al. (2002) h2so4-h2o binary parameterization
+//
+// References:
+// * merikanto, j., i. napari, h. vehkamaki, t. anttila,
+//   and m. kulmala, 2007, new parameterization of
+//   sulfuric acid-ammonia-water ternary nucleation
+//   rates at tropospheric conditions,
+//   j. geophys. res., 112, d15207, doi:10.1029/2006jd0027977
+//
+// * vehkam\"aki, h., m. kulmala, i. napari, k.e.j. lehtinen,
+//   c. timmreck, m. noppel and a. laaksonen, 2002,
+//   an improved parameterization for sulfuric acid-water nucleation
+//   rates for tropospheric and stratospheric conditions,
+//   j. geophys. res., 107, 4622, doi:10.1029/2002jd002184
+//
+// * Wang, M. and J.E. Penner, 2008,
+//   Aerosol indirect forcing in a global model with particle nucleation,
+//   Atmos. Chem. Phys. Discuss., 8, 13943-13998
+//   Atmos. Chem. Phys.  9, 239-260, 2009
+
+// This is the version from the E3SM mam refactor repo
+KOKKOS_INLINE_FUNCTION
+void mer07_veh02_wang08_nuc_1box(
+    // in
+    const int newnuc_method_flagaa, const Real dtnuc, const Real temp_in,
+    const Real rh_in, const Real press_in, const Real zm_in, const Real pblh_in,
+    const Real qh2so4_cur, const Real qh2so4_avg, const Real qnh3_cur,
+    const Real h2so4_uptkrate, const Real mw_so4a_host, const int nsize,
+    // NOTE: in fortran dplom_sect is given/accessed as an array with extent
+    //       maxd_asize defined as "dimension for dplom_sect, ..."
+    //       however, as provided in validation data, it is a scalar
+    // const int maxd_asize,
+    const Real dplom_sect, const Real dphim_sect, const int ldiagaa,
+    // in fortran provided by mo_constants
+    const Real rgas, const Real avogad,
+    // in fortran provided by physconst
+    const Real mw_nh4a, const Real mw_so4a,
+    // in fortran provided by mo_constants
+    const Real pi,
+    //  out
+    int &isize_nuc, Real &qnuma_del, Real &qso4a_del, Real &qnh4a_del,
+    Real &qh2so4_del, Real &qnh3_del, Real &dens_nh4so4a, Real &dnclusterdt) {
+
+  // NOTE: there are a lot of variables that end up unused since we ignore all
+  //       the writing to file at the bottom. for now, I've preserved all of
+  //       the unused bits as comments *just in case*
+
+  // dry-air molar density (mol/m3)
+  // Real cair;
+  // kk2002 "cs_prime" parameter (1/m2)
+  Real cs_prime_kk;
```

Comment:
Should we initialize these variables?

Reply:
Good call--switched to declaring on initialization when first used.

Thanks!